command/resource/client/helper.go

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package client
+
+import (
+	"crypto/tls"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/hashicorp/go-rootcerts"
+)
+
+// tls.Config is used to establish communication in TLS mode
+func SetupTLSConfig(c *GRPCConfig) (*tls.Config, error) {
+	tlsConfig := &tls.Config{
+		InsecureSkipVerify: !c.GRPCTLSVerify,
+	}
+
+	if c.CertFile != "" && c.KeyFile != "" {
+		tlsCert, err := tls.LoadX509KeyPair(c.CertFile, c.KeyFile)
+		if err != nil {
+			return nil, err
+		}
+		tlsConfig.Certificates = []tls.Certificate{tlsCert}
+	}
+
+	var caConfig *rootcerts.Config
+	if c.CAFile != "" || c.CAPath != "" {
+		caConfig = &rootcerts.Config{
+			CAFile: c.CAFile,
+			CAPath: c.CAPath,
+		}
+	}
+	// load system CA certs if user doesn't provide any
+	if err := rootcerts.ConfigureTLS(tlsConfig, caConfig); err != nil {
+		return nil, err
+	}
+
+	return tlsConfig, nil
+}
+
+func removeSchemaFromGRPCAddress(addr string) string {
+	// Parse as host:port with option http prefix
+	grpcAddr := strings.TrimPrefix(addr, "http://")
+	grpcAddr = strings.TrimPrefix(grpcAddr, "https://")
```

Comment:
same here about schema? are we continuing with `http` for the `grpc` schema?

Reply:
As discussed, leave as it is for now.